### PR TITLE
Two key hotfix

### PR DESCRIPTION
**Pull request by Tulun**
-Fix issue with timer. Actually works on fixed time.

-Clear cache on address change.

-Reset keys on change.

### CODE REVIEW

**Review by daywiss — COMMENTED** [review]
looks good, mainly just want to make sure we dont reload page unless absolutley necssary

File: src/features/router/Router.tsx [hunk 1]
```diff
@@ -13,8 +13,34 @@ import Footer from "common/components/footer";
...
+
+  useEffect(() => {
+    if (
+      state.isConnected &&
+      previousAddress &&
+      previousAddress !== state.address
+    ) {
+      window.location.reload();
+    }
+  }, [state.isConnected, state.address, previousAddress]);
```
[the remark this answers is not in the document]
Comment (daywiss) [reply src/features/router/Router.tsx:41]: yea... well it makes all the other changes redundant. we should probably choose one way or the other

[the remark this answers is not in the document]
Comment (Tulun) [reply src/features/router/Router.tsx:41]: I mean I can do it the proper way but that might take me a couple days.

File: src/features/vote/helpers/calculateTimeRemaining.ts [hunk 2]
```diff
@@ -1,20 +1,13 @@
...
-      hours: Math.floor((difference / (60 * 60)) % 24).toString(),
-      minutes: Math.floor((difference / 60) % 60).toString(),
-    };
-    if (Number(timeLeft.hours) < 10) timeLeft.hours = `0${timeLeft.hours}`;
-    if (Number(timeLeft.minutes) < 10)
-      timeLeft.minutes = `0${timeLeft.minutes}`;
-  }
+  let text = "00:00";
+  // format difference
+  if (difference > 0) text = Duration.fromMillis(difference).toFormat("hh:mm");
```
Comment (daywiss) [inline src/features/vote/helpers/calculateTimeRemaining.ts:10]: huge improvement

**Review by daywiss — APPROVED** [review]

**Pull request merged**